### PR TITLE
Added tenant configuration to helm-chart

### PR DESCRIPTION
**Pull request by abhi1693**
In my environment, there is no easy way to use `kubectl`, `kustomize`, or `krew`. However, if the deployment is done via `helm` chart, then I can manage using our automation tools.

Hopefully, this PR will get accepted for merge and would help many others as well to deploy MinIO much easily.

### CODE REVIEW

**Comment by nitisht** [comment]
Thanks @abhi1693 looks interesting. We'll review and test soon.

**Comment by nitisht** [comment]
@abhi1693 There is one issue with the approach here:

The `values.yaml` file doesn't really make it clear that while some parts (operator related) are single / one time operations only. The other parts. i.e. tenant related can be appended with a new tenant configuration every time someone wants to create a new tenant. 

Can we make the tenant part in the values.yaml like a yaml array? Also add some comments on what each part in the file is supposed to do.

**Comment by abhi1693** [comment]
@nitisht I'll update the comments once you ensure that everything else looks good. The values.yaml is loosely based on examples/tenant.yaml file at the moment.

**Review by nitisht — CHANGES_REQUESTED** [review]
@abhi1693 Thanks for the changes. I was trying to deploy but the helm install fails with

```
Error: template: minio-operator/templates/tenant-secret.yaml:8:23: executing "minio-operator/templates/tenant-secret.yaml" at <.Values.tenant.secrets.accessKey>: nil pointer evaluating interface {}.secrets
```

File: helm/minio-operator/templates/tenant.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  ## Refer Kubernetes documentation for details https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#priorityclass/
+  priorityClassName: {{ .priorityClassName }}
+
+  ## Define configuration for Console (Graphical user interface for MinIO)
+  ## Refer https://github.com/minio/console
+  console:
+    image: {{ .console.image.repository }}:{{ .console.image.tag }}
+    replicas: {{ .console.replicaCount }}
+    consoleSecret:
+      name: console-secret
```
Comment (nitisht) [inline helm/minio-operator/templates/tenant.yaml:152]: Console-secret should be a user input as well.

File: helm/minio-operator/templates/tenant.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    {{- with .securityContext }}
+    securityContext:
+      {{ toYaml . | nindent 6 }}
+    {{ end }}
+{{ end }}
```
Comment (nitisht) [inline helm/minio-operator/templates/tenant.yaml:157]: newline

File: helm/minio-operator/values.yaml [hunk 3]
```diff
@@ -23,3 +23,52 @@ console:
...
+    podManagementPolicy: Parallel
+    serviceMetadata: {}
+    env: {}
+    priorityClassName : high-priority
+    console:
+      image:
+        repository: minio/console
+        tag: v0.6.3
+        pullPolicy: IfNotPresent
+      replicaCount: 1
```
Comment (nitisht) [inline helm/minio-operator/values.yaml:74]: newline

File: helm/minio-operator/values.yaml [hunk 4]
```diff
@@ -23,3 +23,52 @@ console:
...
+      enabled: false
+      port: 9000
+    pools:
+      servers: 4
+      volumesPerServer: 4
+    certificate:
+      externalCertSecret: {}
+      requestAutoCert: true
+      certConfig: {}
+      s3:
```
Comment (nitisht) [inline helm/minio-operator/values.yaml:63]: `s3` can be a top level field, i.e. at par with `certificate`

File: helm/minio-operator/values.yaml [hunk 5]
```diff
@@ -23,3 +23,52 @@ console:
     tag: v0.6.3
     pullPolicy: IfNotPresent
   replicaCount: 1
+  secrets:
```
Comment (nitisht) [inline helm/minio-operator/values.yaml:26]: This console section is for operator console. These secrets have to be added to a new `console` section under tenants

File: helm/minio-operator/templates/console-secret.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+type: Opaque
+data:
+  ## Passphrase to encrypt jwt payload
+  CONSOLE_PBKDF_PASSPHRASE: {{ .Values.console.secrets.passphrase }}
+  ## Salt to encrypt jwt payload
+  CONSOLE_PBKDF_SALT: {{ .Values.console.secrets.salt }}
+  ## MinIO User Access Key (used for Console Login)
+  CONSOLE_ACCESS_KEY: {{ .Values.console.secrets.accessKey }}
+  ## MinIO User Secret Key (used for Console Login)
+  CONSOLE_SECRET_KEY: {{ .Values.console.secrets.secretKey }}
```
Comment (nitisht) [inline helm/minio-operator/templates/console-secret.yaml:14]: newline

File: helm/minio-operator/templates/tenant-secret.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: minio-creds-secret
+type: Opaque
+data:
+  ## Access Key for MinIO Tenant
+  accesskey: {{ .Values.tenant.secrets.accessKey }}
+  ## Secret Key for MinIO Tenant
+  secretkey: {{ .Values.tenant.secrets.secretkey }}
```
Comment (nitisht) [inline helm/minio-operator/templates/tenant-secret.yaml:10]: newline

**Comment by abhi1693** [comment]
@nitisht I have a suggestion to avoid future template deploy failures. Can you add a GitHub action workflow that uses `kubeval` and `helm template` commands to verify the configuration?

The command is: 
```
helm template --values "$chart_path/values.yaml" "$chart_path" | kubeval --strict --exit-on-error --output tap
```

**Comment by abhi1693** [comment]
@nitisht Can you also add another GitHub workflow to update the documentation automatically for any changed value in values.yaml using `helm-docs`?

**Comment by nitisht** [comment]
Have you tested this @abhi1693 ? I see this error when running it 

```
$ helm install ./minio-operator/ --generate-name --create-namespace --namespace minio-operator
WARNING: "kubernetes-charts.storage.googleapis.com" is deprecated for "stable" and will be deleted Nov. 13, 2020.
WARNING: You should switch to "https://charts.helm.sh/stable" via:
WARNING: helm repo add "stable" "https://charts.helm.sh/stable" --force-update
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Tenant" in version "minio.min.io/v2"

```

**Comment by abhi1693** [comment]
@nitisht I have fixed the issue for now, but you will have to make sure that the crd is always placed under `crds` directory. But this will again not work if the file is placed by your team under `templates`

**Comment by nitisht** [comment]
There are some more fixes to be done, I'll send a PR to your fork. We can proceed from there

**Comment by nitisht** [comment]
can you take this https://github.com/abhi1693/operator/pull/1 @abhi1693 ?

**Review by nitisht — APPROVED** [review]
LGTM, basic testing working.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**